File: 2022/day4/ocaml/bin/main.ml
open Aoc2022day4

let () =
    let part, path = match Common.get_opt Sys.argv 1, Common.get_opt Sys.argv 2 with
    | Some "1", Some path -> 1, path
    | Some _, Some path -> 2, path
    | Some "1", None -> 1, "../input.txt"
    | _ -> 2, "../input.txt" in

    let solver = if part == 1 then Part1.solve else Part2.solve in
    let ic = open_in path in
    try
        let answer = solver 0 ic in
        print_int answer;
        print_newline ();
        close_in ic
    with e ->
        close_in_noerr ic;
        raise e

File: 2022/day4/rust/src/main.rs
use aoc22d4::*;
use std::{fs::File, io::BufReader};

fn main() {
    let (part, path) = parse_args();
    let path = path.unwrap_or_else(|| String::from("../input.txt"));
    let part = match part.as_deref() {
        Some("help") | Some("--help") => {
            println!("USAGE:\naoc22d4 [part number] [path]");
            return;
        }
        Some("1") => 1,
        _ => 2,
    };

    let file = File::open(&path).unwrap_or_else(|_| panic!("Couldn't open file at path {}", path));
    let answer = if part == 1 {
        part1::solve(BufReader::new(file))
    } else {
        part2::solve(BufReader::new(file))
    };
    println!("{answer}");
}
